openai/src/commonMain/kotlin/com/xebia/functional/xef/auto/llm/openai/MockAIClient.kt

```diff
@@ -1,131 +0,0 @@
-
-  override suspend fun createCompletion(request: CompletionRequest): CompletionResult =
-    completion(request)
-
-  override suspend fun createChatCompletion(
-    request: ChatCompletionRequest
-  ): ChatCompletionResponse = chatCompletion(request)
-
-  override suspend fun createChatCompletionWithFunctions(
-    request: ChatCompletionRequestWithFunctions
-  ): ChatCompletionResponseWithFunctions = chatCompletionRequestWithFunctions(request)
-
-  override suspend fun createChatCompletions(
-    request: ChatCompletionRequest
-  ): Flow<ChatCompletionChunk> = chatCompletions(request)
-
-  override suspend fun createEmbeddings(request: EmbeddingRequest): EmbeddingResult =
-    embeddings(request)
-
-  override suspend fun createImages(request: ImagesGenerationRequest): ImagesGenerationResponse =
-    images(request)
-
-  override fun tokensFromMessages(messages: List<Message>): Int = 0
-
-  override fun close() {}
-}
-
-fun nullEmbeddings(request: EmbeddingRequest): EmbeddingResult {
-  val results = request.input.mapIndexed { index, s -> Embedding(s, listOf(0F), index) }
-  return EmbeddingResult(results, Usage.ZERO)
-}
-
-fun simpleMockAIClient(execute: (String) -> String): MockOpenAIClient =
-  MockOpenAIClient(
-    completion = { req ->
-      val request = "${req.prompt.orEmpty()} ${req.suffix.orEmpty()}"
-      val response = execute(request)
-      val result = CompletionChoice(response, 0, null, "end")
-      val requestTokens = request.split(' ').size
-      val responseTokens = response.split(' ').size
-      val usage = Usage(requestTokens, responseTokens, requestTokens + responseTokens)
-      CompletionResult("FakeID123", "", 0, req.model, listOf(result), usage)
-    },
-    chatCompletion = { req ->
-      val responses =
-        req.messages.mapIndexed { ix, msg ->
-          val response = execute(msg.content ?: "")
-          Choice(Message(msg.role, response, msg.role.name), "end", ix)
-        }
-      val requestTokens = req.messages.sumOf { it.content?.split(' ')?.size ?: 0 }
-      val responseTokens = responses.sumOf { it.message?.content?.split(' ')?.size ?: 0 }
-      val usage = Usage(requestTokens, responseTokens, requestTokens + responseTokens)
-      ChatCompletionResponse("FakeID123", "", 0, req.model, usage, responses)
-    }
-  )
-
-@OptIn(ExperimentalTime::class)
-suspend fun <A> MockAIScope(
-  mockClient: MockOpenAIClient,
-  block: suspend Conversation.() -> A,
-  orElse: suspend (AIError) -> A
-): A =
-  try {
-    val embeddings = OpenAIEmbeddings(mockClient)
-    val vectorStore = LocalVectorStore(embeddings)
-    val scope = Conversation(embeddings, vectorStore)
-    block(scope)
-  } catch (e: AIError) {
-    orElse(e)
-  }
-
-/**
- * Run the [AI] value to produce _either_ an [AIError], or [A]. This method uses the [mockAI] to
- * compute the different responses.
- */
-suspend fun <A> AI<A>.mock(mockAI: MockOpenAIClient): Either<AIError, A> =
```

Comment:
I don't see the value of this at the moment, and I removed it because it was not in use anywhere.
We should have a similar test suite for conversations like the one you did for message token count.